drivers/Watchdog.h

```diff
@@ -20,21 +20,24 @@
 
 #ifdef DEVICE_WATCHDOG
 
-#include "watchdog_api.h"
-
 #include <cstdio>
+#include "mbed_error.h"
+#include "rtos/ThisThread.h"
+#include "platform/mbed_critical.h"
+#include "platform/mbed_power_mgmt.h"
+#include <string>
+namespace mbed
+{
 
-
-namespace mbed {
 /** \addtogroup drivers */
 /** A system timer that will reset the system in the case of system failures or
  *  malfunctions.
  *
  * Example:
  * @code
  *
- * Watchdog watchdog = Watchdog();
- * watchdog.start(2000);
+ * Watchdog watchdog = Watchdog(300,"Software Watchdog");
```

Comment:
You may want to do direct initialization rather than assign. Something like `Watchdog watchdog(300,"Software Watchdog");`.